src/main/java/org/openstreetmap/atlas/checks/validation/relations/InvalidMultiPolygonRelationCheck.java

```diff
@@ -185,43 +161,254 @@ else if (!relationMember.getRole().equals(RelationTypeTag.MULTIPOLYGON_ROLE_INNE
+     *            {@link Relation} of type multipolygon
+     * @return an Optional containing a {@link Tuple} containing a {@link Set} of {@link String}
+     *         instructions for invalid geometries and a {@link Set} of {@link Location}s marking
+     *         the invalid geometries
+     */
+    private Optional<Tuple<Set<String>, Set<Location>>> checkGeometry(
+            final Relation multipolygonRelation)
     {
+        // Try converting the Relation to a MultiPolygon. If it works check that the geometries
+        // don't overlap.
         try
         {
-            RELATION_OR_AREA_TO_MULTI_POLYGON_CONVERTER.convert(relation);
+            return Optional.of(checkOverlap(
+                    RELATION_OR_AREA_TO_MULTI_POLYGON_CONVERTER.convert(multipolygonRelation),
+                    multipolygonRelation.getOsmIdentifier()));
         }
+        // Catch open polygons and mark the broken locations
         catch (final MultiplePolyLineToPolygonsConverter.OpenPolygonException exception)
         {
             final List<Location> openLocations = exception.getOpenLocations();
             final Set<AtlasObject> objects = openLocations.stream()
-                    .flatMap(location -> this.filterMembers(relation, location))
+                    .flatMap(location -> this.filterMembers(multipolygonRelation, location))
                     .collect(Collectors.toSet());
-            final Set<Long> memberIds = relation.members().stream()
+            final Set<Long> memberIds = multipolygonRelation.members().stream()
                     .map(member -> member.getEntity().getOsmIdentifier())
                     .collect(Collectors.toSet());
 
             if (!objects.isEmpty() && !memberIds.isEmpty())
             {
-                return Optional
-                        .of(this.getLocalizedInstruction(CLOSED_LOOP_INSTRUCTION_FORMAT_INDEX,
-                                relation.getOsmIdentifier(), memberIds, openLocations));
+                return Optional.of(Tuple.createTuple(
+                        Collections.singleton(this.getLocalizedInstruction(
+                                CLOSED_LOOP_INSTRUCTION_FORMAT_INDEX,
+                                multipolygonRelation.getOsmIdentifier(), memberIds, openLocations)),
+                        new HashSet<>(openLocations)));
             }
             else
             {
-                logger.warn("Unable to find members in relation {} containing the locations : {}",
-                        relation, openLocations);
+                logger.warn(
+                        "Unable to find members in multipolygonRelation {} containing the locations : {}",
+                        multipolygonRelation, openLocations);
+            }
+        }
+        catch (final CoreException exception)
+        {
+            // Catch multipolygon relations with no outer members
+            if (exception.getMessage().equals("Unable to find outer polygon."))
```

Comment:
Not sure if this is the best way to assign instructions for the different cases. The instructions could get changed in the future. Something to look out for.

Reply:
It's not ideal, but if it does change it should be caught by the unit tests.